### PR TITLE
Proofread and update documentation for *_join functions

### LINKED ISSUE #4787: Revamp `*_join()` docs
* [ ] Proofread title, description, and parameters
* [ ] Proofread comments in examples
* [ ] Ensure comment headings used to divide into scannable sections
* [ ] Ensure there's one example that shows tidy eval and links to `?dplyr_tidy_eval`
* [ ] Ensure there's one example that shows `across()`
* [ ] Flag examples that seem overly complex for further review
* [ ] Switch out mtcars/iris for something more interesting, or a smaller dataset constructed specifically to illustrate one point. 
* [ ] Review size of example outputs to ensure that you can easily see important rows and cols.
* [ ] Datasets should always be tibbles to avoid use of `as_tibble()` distracting from the main point

### PR DESCRIPTION
**Pull request by davisadamw**
Fixes #4787

### CODE REVIEW

File: R/join.r [hunk 1]
```diff
@@ -9,27 +9,27 @@
 #' * `right_join()`: includes all rows in `y`.
 #' * `full_join()`: includes all rows in `x` or `y`.
 #'
-#' If there are multiple matches between `x` and `y`, all combination of the
+#' If there are multiple matches between rows in `x` and `y`, all combinations of the
```
Comment (hadley) [inline R/join.r]: I recently realised it's slightly simpler to say: If a row in `x` matches multiple rows in `y`, all the rows in `y` will be returned. Does that make sense?

Comment (davisadamw) [reply R/join.r]: That is simpler and describes the behavior accurately. I think it's still helpful to make it clear that the number of rows returned from each multiple match is [matches from X] * [matches from Y] that set of match, so maybe something like: If a row in `x` matches multiple rows in `y`, all the rows in `y` will be returned once for each matching row in `x`

Comment (hadley) [reply R/join.r]: Sounds good!

Comment (davisadamw) [reply R/join.r]: Great, thanks!

**Pull request merged**